From b9fc71246e9f217f51d60f5482d8bcbb20e73a39 Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Fri, 20 Apr 2018 23:46:22 +0800
Subject: [PATCH 3680/5242] MLK-18101-4 gpu: imx: dpu-blit: Refine the first
 frame irq handle

commit  3739c08cede6117d98379a734610ce4ac139318c from
https://source.codeaurora.org/external/imx/linux-imx.git

As the display will also use dprc0-irq, which will disturb the
interrupt for blit engine if blit engine use irq.
Refine the first frame process so that blit engine never use irq any more.

Signed-off-by: yuchou gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu-blit/dpu-blit.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dpu-blit/dpu-blit.c b/drivers/gpu/imx/dpu-blit/dpu-blit.c
index 7035dd0..bac49db 100644
--- a/drivers/gpu/imx/dpu-blit/dpu-blit.c
+++ b/drivers/gpu/imx/dpu-blit/dpu-blit.c
@@ -123,7 +123,7 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	dpu_be_wait(dpu_be);
 
 	if (need_handle_start) {
-		dprc_irq_handle(dprc);
+		dprc_first_frame_handle(dprc);
 		need_handle_start = false;
 	}
 
@@ -141,7 +141,6 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	dprc_reg_update(dprc);
 
 	if (start) {
-		dprc_enable_ctrl_done_irq(dprc);
 		need_handle_start = true;
 	}
 
-- 
1.7.9.5

